apps/api/src/domain/factories/__tests__/aggregate-factories.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { NotImplementedException } from '@nestjs/common';
+import { InternalError } from '../../../lib/errors/InternalError';
+import { DTO } from '../../../types/DTO';
+import formatAggregateType from '../../../view-models/presentation/formatAggregateType';
+import { Aggregate } from '../../models/aggregate.entity';
+import { AggregateType } from '../../types/AggregateType';
+import { CategorizableType } from '../../types/CategorizableType';
+import { isResourceType } from '../../types/ResourceType';
+import getInstanceFactoryForResource from '../getInstanceFactoryForResource';
+import buildAggregateFactoryTestCases from './buildAggregateFactoryTestCases';
+
+const testCaseSets = buildAggregateFactoryTestCases();
+
+const buildInstanceFactoryFunction = (aggregateType: AggregateType) => (dto: DTO<Aggregate>) =>
```

Comment:
Looks like this should be production code. In fact, this is the thing that we are testing!

Reply:
I have removed the extra layer.